sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -37,7 +46,115 @@ def test_transform_linear_model():
 
 def test_invalid_input():
     clf = SGDClassifier(alpha=0.1, n_iter=10, shuffle=True, random_state=None)
+    for threshold in ["gobbledigook", ".5 * gobbledigook"]:
+        model = SelectFromModel(clf, threshold=threshold)
+        model.fit(iris.data, iris.target)
+        assert_raises(ValueError, model.transform, iris.data)
+
+
+def test_validate_estimator():
+    est = RandomForestClassifier()
+    transformer = SelectFromModel(estimator=est)
+    transformer.fit(iris.data, iris.target)
+    assert_equal(transformer.estimator, est)
+
+
+def test_feature_importances():
+    X, y = datasets.make_classification(n_samples=1000,
+                                        n_features=10,
+                                        n_informative=3,
+                                        n_redundant=0,
+                                        n_repeated=0,
+                                        shuffle=False,
+                                        random_state=0)
+
+    est = RandomForestClassifier(n_estimators=50, random_state=0)
+    transformer = SelectFromModel(estimator=est)
+
+    transformer.fit(X, y)
+    assert_true(hasattr(transformer.estimator_, 'feature_importances_'))
+
+    X_new = transformer.transform(X)
+    assert_less(X_new.shape[1], X.shape[1])
+
+    feature_mask = (transformer.estimator_.feature_importances_ >
+                    transformer.estimator_.feature_importances_.mean())
+    assert_array_almost_equal(X_new, X[:, feature_mask])
+
+    # Check with sample weights
+    sample_weight = np.ones(y.shape)
+    sample_weight[y == 1] *= 100
+
+    est = RandomForestClassifier(n_estimators=50, random_state=0)
+    transformer = SelectFromModel(estimator=est)
+    transformer.fit(X, y, sample_weight=sample_weight)
+    importances = transformer.estimator_.feature_importances_
+    assert_less(importances[1], X.shape[1])
```

Comment:
What does this test? And isn't that testing just something in the forest unspecific to the SelectFromModel?